### PR TITLE
Promote speed of function "clustering" and "tracking"

### PR DESCRIPTION
**Pull request by chen1i6c04**

### CODE REVIEW

File: src/utils/data.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import pandas as pd
+
+
+def convert_data_type(profile):
```
Comment (yuehhua) [inline src/utils/data.py]:
```suggestion
def integer_encoding(profile):
```

File: src/algorithms/clustering.py [hunk 2]
```diff
@@ -7,12 +7,13 @@
 import matplotlib.pyplot as plt
 from matplotlib.ticker import MaxNLocator, FuncFormatter
 
+from ..utils.data import convert_data_type
```
Comment (yuehhua) [inline src/algorithms/clustering.py]:
```suggestion
from ..utils.data import integer_encoding
```

File: src/algorithms/clustering.py [hunk 3]
```diff
@@ -7,12 +7,13 @@
...
+
 
 class DistanceMatrix:
 
     def __init__(self, profile):
-        profile = profile.fillna('0')
-        profile = profile.transpose()
+        profile = convert_data_type(profile).T
```
Comment (yuehhua) [inline src/algorithms/clustering.py]:
```suggestion
        profile = integer_encoding(profile).T
```

File: src/algorithms/bionumerics.py [hunk 4]
```diff
@@ -1,17 +1,7 @@
...
-        allele_set = set(value)
-        allele_set.discard(0)
-        encoder = {allele: index for index, allele in enumerate(allele_set, 1)}
-        value = [encoder.get(i, 0) for i in value]
-        new_values.append(value)
-    df = pd.DataFrame(data=new_values, index=profile.index, columns=profile.columns)
-    df = df.transpose()
-    df.insert(0, 'Key', df.index)
-    return df
+    profile = convert_data_type(profile).T
```
Comment (yuehhua) [inline src/algorithms/bionumerics.py]:
```suggestion
    profile = integer_encoding(profile).T
```

File: src/algorithms/bionumerics.py [hunk 5]
```diff
@@ -1,17 +1,7 @@
-import pandas as pd
+from ..utils.data import convert_data_type
```
Comment (yuehhua) [inline src/algorithms/bionumerics.py]:
```suggestion
from ..utils.data import integer_encoding
```

**Comment by yuehhua** [comment]
LGTM

**Comment by yi-syong** [comment]
GOOD!

**Pull request merged**